File: java/trunk/TetramBDEJB/ejbModule/org/tetram/bdtheque/model/entity/Edition.java
package org.tetram.bdtheque.model.entity;

import java.io.Serializable;
import java.math.BigDecimal;
import java.sql.Blob;
import java.util.Date;
import java.util.List;

import javax.persistence.Column;
import javax.persistence.Embedded;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.Lob;
import javax.persistence.ManyToOne;
import javax.persistence.OneToMany;
import javax.persistence.OrderColumn;
import javax.persistence.Table;
import javax.persistence.Temporal;
import javax.persistence.TemporalType;

@Entity
@Table(name = "editions")
public class Edition implements Serializable {

	/**
	 * 
	 */
	private static final long serialVersionUID = 1L;

	@GeneratedValue
	@Id
	@Column(name = "id_edition")
	private String id;

	@ManyToOne(optional = false, fetch = FetchType.LAZY)
	@JoinColumn(name = "id_album")
	private Album album;

	@ManyToOne(optional = false, fetch = FetchType.LAZY)
	@JoinColumn(name = "id_editeur")
	private Editeur editeur;
	@ManyToOne(optional = true, fetch = FetchType.LAZY)
	@JoinColumn(name = "id_collection")
	private Collection collection;

	private Integer anneeEdition;
	@Column(columnDefinition = "numeric(15,2)")
	private BigDecimal prix;
	private Boolean vo;
	private Boolean couleur;
	private String isbn;
	private Boolean prete;
	private Boolean stock;
	private Boolean dedicace;
	@Temporal(TemporalType.DATE)
	private Date dateAchat;
	private Boolean gratuit;
	private Boolean offert;
	@Column(name = "nombredepages")
	private Integer nbPages;

	@Embedded
	private ConfigurationEdition etat;

	@Embedded
	private Cote cote;
	private String numeroPerso;
	@Lob
	@Column(name = "notes")
	private Blob commentaires;

	@OneToMany(mappedBy = "edition", fetch = FetchType.LAZY)
	@OrderColumn(name = "anneecote")
	private List<CoteEdition> cotes;

	@OneToMany(mappedBy = "edition", fetch = FetchType.LAZY)
	@OrderColumn(name = "ordre")
	private List<EditionImage> images;

	public String getId() {
		return id;
	}

	public void setId(String id) {
		this.id = id;
	}

	public Album getAlbum() {
		return album;
	}

	public void setAlbum(Album album) {
		this.album = album;
	}

	public Editeur getEditeur() {
		return editeur;
	}

	public void setEditeur(Editeur editeur) {
		this.editeur = editeur;
	}

	public Collection getCollection() {
		return collection;
	}

	public void setCollection(Collection collection) {
		this.collection = collection;
	}

	public Integer getAnneeEdition() {
		return anneeEdition;
	}

	public void setAnneeEdition(Integer anneeEdition) {
		this.anneeEdition = anneeEdition;
	}

	public BigDecimal getPrix() {
		return prix;
	}

	public void setPrix(BigDecimal prix) {
		this.prix = prix;
	}

	public Boolean getVo() {
		return vo;
	}

	public void setVo(Boolean vo) {
		this.vo = vo;
	}

	public Boolean getCouleur() {
		return couleur;
	}

	public void setCouleur(Boolean couleur) {
		this.couleur = couleur;
	}

	public String getIsbn() {
		return isbn;
	}

	public void setIsbn(String isbn) {
		this.isbn = isbn;
	}

	public Boolean getPrete() {
		return prete;
	}

	public void setPrete(Boolean prete) {
		this.prete = prete;
	}

	public Boolean getStock() {
		return stock;
	}

	public void setStock(Boolean stock) {
		this.stock = stock;
	}

	public Boolean getDedicace() {
		return dedicace;
	}

	public void setDedicace(Boolean dedicace) {
		this.dedicace = dedicace;
	}

	public Date getDateAchat() {
		return dateAchat;
	}

	public void setDateAchat(Date dateAchat) {
		this.dateAchat = dateAchat;
	}

	public Boolean getGratuit() {
		return gratuit;
	}

	public void setGratuit(Boolean gratuit) {
		this.gratuit = gratuit;
	}

	public Boolean getOffert() {
		return offert;
	}

	public void setOffert(Boolean offert) {
		this.offert = offert;
	}

	public Integer getNbPages() {
		return nbPages;
	}

	public void setNbPages(Integer nbPages) {
		this.nbPages = nbPages;
	}

	public ConfigurationEdition getEtat() {
		return etat;
	}

	public void setEtat(ConfigurationEdition etat) {
		this.etat = etat;
	}

	public Cote getCote() {
		return cote;
	}

	public void setCote(Cote cote) {
		this.cote = cote;
	}

	public String getNumeroPerso() {
		return numeroPerso;
	}

	public void setNumeroPerso(String numeroPerso) {
		this.numeroPerso = numeroPerso;
	}

	public Blob getCommentaires() {
		return commentaires;
	}

	public void setCommentaires(Blob commentaires) {
		this.commentaires = commentaires;
	}

	public List<CoteEdition> getCotes() {
		return cotes;
	}

	public void setCotes(List<CoteEdition> cotes) {
		this.cotes = cotes;
	}

	public List<EditionImage> getImages() {
		return images;
	}

	public void setImages(List<EditionImage> images) {
		this.images = images;
	}

}

File: php/trunk/src/BDTheque/DataModelBundle/Entity/Edition.php
<?php

namespace BDTheque\DataModelBundle\Entity;

use Doctrine\ORM\Mapping as ORM;

/**
 * BDTheque\DataModelBundle\Entity\Edition
 *
 * @ORM\Table(name="editions")
 * @ORM\Entity(repositoryClass="BDTheque\DataModelBundle\Entity\EditionRepository")
 */
class Edition extends BaseEntity {

    /**
     * @var boolean $stock
     *
     * @ORM\Column(name="stock", type="boolean")
     */
    private $stock;

    /**
     * @var Image[] $couvertures
     *
     * @ORM\OneToMany(targetEntity="Image", mappedBy="edition", cascade={"persist"}, orphanRemoval=true)
     */
    private $images;

    /**
     * @var float $prix
     *
     * @ORM\Column(name="prix", type="decimal", precision=2, nullable=true)
     */
    private $prix;

    /**
     * @var integer $anneeCote
     *
     * @ORM\Column(name="anneeCote", type="integer", nullable=true)
     */
    private $anneeCote;

    /**
     * @var string $isbn
     *
     * @ORM\Column(name="isbn", type="string", length=255, nullable=true)
     */
    private $isbn;

    /**
     * @var boolean $gratuit
     *
     * @ORM\Column(name="gratuit", type="boolean")
     */
    private $gratuit;

    /**
     * @var boolean $prete
     *
     * @ORM\Column(name="prete", type="boolean")
     */
    private $prete;

    /**
     * @var integer $nombreDePages
     *
     * @ORM\Column(name="nombreDePages", type="integer", nullable=true)
     */
    private $nombreDePages;

    /**
     * @var string $numeroPerso
     *
     * @ORM\Column(name="numeroPerso", type="string", length=255, nullable=true)
     */
    private $numeroPerso;

    /**
     * @var string $notes
     *
     * @ORM\Column(name="notes", type="text", nullable=true)
     */
    private $notes;

    /**
     * @var integer $anneeEdition
     *
     * @ORM\Column(name="anneeEdition", type="integer", nullable=true)
     */
    private $anneeEdition;

    /**
     * @var boolean $dedicace
     *
     * @ORM\Column(name="dedicace", type="boolean")
     */
    private $dedicace;

    /**
     * @var boolean $couleur
     *
     * @ORM\Column(name="couleur", type="boolean")
     */
    private $couleur;

    /**
     * @var Collection $collection
     *
     * @ORM\ManyToOne(targetEntity="Collection", cascade={"all"}, fetch="LAZY")
     * @ORM\JoinColumn(nullable=true)
     */
    private $collection;

    /**
     * @var float $prixCote
     *
     * @ORM\Column(name="prixCote", type="decimal", precision=2, nullable=true)
     */
    private $prixCote;

    /**
     * @var \DateTime $dateAchat
     *
     * @ORM\Column(name="dateAchat", type="date", nullable=true)
     */
    private $dateAchat;

    /**
     * @var boolean $offert
     *
     * @ORM\Column(name="offert", type="boolean")
     */
    private $offert;

    /**
     * @var Editeur $editeur
     *
     * @ORM\ManyToOne(targetEntity="Editeur", cascade={"all"}, fetch="LAZY")
     * @ORM\JoinColumn(nullable=false)
     */
    private $editeur;

    /**
     * @var Album $album
     *
     * @ORM\ManyToOne(targetEntity="Album", cascade={"all"}, fetch="LAZY", inversedBy="editions")
     * @ORM\JoinColumn(nullable=false)
     */
    private $album;

    /**
     * @var boolean $vo
     *
     * @ORM\Column(name="vo", type="boolean")
     */
    private $vo;

    public function __construct() {
        $this->images = new \Doctrine\Common\Collections\ArrayCollection();
    }

    /**
     * Set stock
     *
     * @param boolean $stock
     * @return Edition
     */
    public function setStock($stock) {
        $this->stock = $stock;

        return $this;
    }

    /**
     * Get stock
     *
     * @return boolean 
     */
    public function getStock() {
        return $this->stock;
    }

    /**
     * Set prix
     *
     * @param float $prix
     * @return Edition
     */
    public function setPrix($prix) {
        $this->prix = $prix;

        return $this;
    }

    /**
     * Get prix
     *
     * @return float 
     */
    public function getPrix() {
        return $this->prix;
    }

    /**
     * Set anneeCote
     *
     * @param integer $anneeCote
     * @return Edition
     */
    public function setAnneeCote($anneeCote) {
        $this->anneeCote = $anneeCote;

        return $this;
    }

    /**
     * Get anneeCote
     *
     * @return integer 
     */
    public function getAnneeCote() {
        return $this->anneeCote;
    }

    /**
     * Set isbn
     *
     * @param string $isbn
     * @return Edition
     */
    public function setIsbn($isbn) {
        $this->isbn = $isbn;

        return $this;
    }

    /**
     * Get isbn
     *
     * @return string 
     */
    public function getIsbn() {
        return $this->isbn;
    }

    /**
     * Set gratuit
     *
     * @param boolean $gratuit
     * @return Edition
     */
    public function setGratuit($gratuit) {
        $this->gratuit = $gratuit;

        return $this;
    }

    /**
     * Get gratuit
     *
     * @return boolean 
     */
    public function getGratuit() {
        return $this->gratuit;
    }

    /**
     * Set prete
     *
     * @param boolean $prete
     * @return Edition
     */
    public function setPrete($prete) {
        $this->prete = $prete;

        return $this;
    }

    /**
     * Get prete
     *
     * @return boolean 
     */
    public function getPrete() {
        return $this->prete;
    }

    /**
     * Set nombreDePages
     *
     * @param integer $nombreDePages
     * @return Edition
     */
    public function setNombreDePages($nombreDePages) {
        $this->nombreDePages = $nombreDePages;

        return $this;
    }

    /**
     * Get nombreDePages
     *
     * @return integer 
     */
    public function getNombreDePages() {
        return $this->nombreDePages;
    }

    /**
     * Set numeroPerso
     *
     * @param string $numeroPerso
     * @return Edition
     */
    public function setNumeroPerso($numeroPerso) {
        $this->numeroPerso = $numeroPerso;

        return $this;
    }

    /**
     * Get numeroPerso
     *
     * @return string 
     */
    public function getNumeroPerso() {
        return $this->numeroPerso;
    }

    /**
     * Set notes
     *
     * @param string $notes
     * @return Edition
     */
    public function setNotes($notes) {
        $this->notes = $notes;

        return $this;
    }

    /**
     * Get notes
     *
     * @return string 
     */
    public function getNotes() {
        return $this->notes;
    }

    /**
     * Set anneeEdition
     *
     * @param integer $anneeEdition
     * @return Edition
     */
    public function setAnneeEdition($anneeEdition) {
        $this->anneeEdition = $anneeEdition;

        return $this;
    }

    /**
     * Get anneeEdition
     *
     * @return integer 
     */
    public function getAnneeEdition() {
        return $this->anneeEdition;
    }

    /**
     * Set dedicace
     *
     * @param boolean $dedicace
     * @return Edition
     */
    public function setDedicace($dedicace) {
        $this->dedicace = $dedicace;

        return $this;
    }

    /**
     * Get dedicace
     *
     * @return boolean 
     */
    public function getDedicace() {
        return $this->dedicace;
    }

    /**
     * Set couleur
     *
     * @param boolean $couleur
     * @return Edition
     */
    public function setCouleur($couleur) {
        $this->couleur = $couleur;

        return $this;
    }

    /**
     * Get couleur
     *
     * @return boolean 
     */
    public function getCouleur() {
        return $this->couleur;
    }

    /**
     * Set collection
     *
     * @param BDTheque\DataModelBundle\Entity\Collection $collection
     * @return Edition
     */
    public function setCollection(Collection $collection) {
        $this->collection = $collection;

        return $this;
    }

    /**
     * Get collection
     *
     * @return BDTheque\DataModelBundle\Entity\Collection
     */
    public function getCollection() {
        return $this->collection;
    }

    /**
     * Set prixCote
     *
     * @param float $prixCote
     * @return Edition
     */
    public function setPrixCote($prixCote) {
        $this->prixCote = $prixCote;

        return $this;
    }

    /**
     * Get prixCote
     *
     * @return float 
     */
    public function getPrixCote() {
        return $this->prixCote;
    }

    /**
     * Set dateAchat
     *
     * @param \DateTime $dateAchat
     * @return Edition
     */
    public function setDateAchat(\DateTime $dateAchat) {
        $this->dateAchat = $dateAchat;

        return $this;
    }

    /**
     * Get dateAchat
     *
     * @return \DateTime 
     */
    public function getDateAchat() {
        return $this->dateAchat;
    }

    /**
     * Set offert
     *
     * @param boolean $offert
     * @return Edition
     */
    public function setOffert($offert) {
        $this->offert = $offert;

        return $this;
    }

    /**
     * Get offert
     *
     * @return boolean 
     */
    public function getOffert() {
        return $this->offert;
    }

    /**
     * Set editeur
     *
     * @param BDTheque\DataModelBundle\Entity\Editeur $editeur
     * @return Edition
     */
    public function setEditeur(Editeur $editeur) {
        $this->editeur = $editeur;

        return $this;
    }

    /**
     * Get editeur
     *
     * @return BDTheque\DataModelBundle\Entity\Editeur 
     */
    public function getEditeur() {
        return $this->editeur;
    }

    /**
     * Set album
     *
     * @param BDTheque\DataModelBundle\Entity\Album $album
     * @return Edition
     */
    public function setAlbum(Album $album) {
        if ($album == $this->album)
            return $this;

        if ($this->album != null)
            $this->album->removeEdition($this);
        $this->album = $album;
        if ($album != null)
            $album->addEdition($this);

        return $this;
    }

    /**
     * Get album
     *
     * @return Album
     */
    public function getAlbum() {
        return $this->album;
    }

    /**
     * Set vo
     *
     * @param boolean $vo
     * @return Edition
     */
    public function setVo($vo) {
        $this->vo = $vo;

        return $this;
    }

    /**
     * Get vo
     *
     * @return boolean 
     */
    public function getVo() {
        return $this->vo;
    }

    /**
     * Add image
     *
     * @param BDTheque\DataModelBundle\Entity\Image $image
     * @return Edition
     */
    public function addImage(Image $image) {
        if (!$this->images->contains($image)) {
            $this->images[] = $image;
            $image->setEdition($this);
        }

        return $this;
    }

    /**
     * Remove image
     *
     * @param BDTheque\DataModelBundle\Entity\Image $image
     */
    public function removeImage(Image $image) {
        if ($this->images->contains($image)) {
            $this->images->removeElement($image);
            $image->setEdition(null);
        }
    }

    /**
     * Get images
     *
     * @return Doctrine\Common\Collections\Collection 
     */
    public function getImages() {
        return $this->images;
    }

}